### DISCUSSION TITLE
Exclude specific IP from traffic selector

### DISCUSSION THREAD

**Discussion by ptchuba** [strongswan/strongswan#1687] [Q&A] [answered]
I have a roadwarrior setup that works fine. Server is Ubuntu 22.04 and clients are Ubuntu 22.04 and Windows server 2022. The clients are virtual machines on public clouds - mainly Azure and are currently using split-tunneling to only send specific traffic through the VPN. 

I now have a requirement to route many more networks over the VPN and it has become too cumbersome to list all of these in the remote_ts or in the case of Windows to create routes for all of them. I've therefore decided to use full-tunneling. However, because these boxes are on the cloud, I need to access them through ssh or rdp. The issue I'm having now is that when I connect to the linux client through ssh and then initiate the VPN (with remote_ts = 0.0.0.0/0) I lose access to the client immediately the VPN connects. I understand this happens because the return traffic to my machine is now tunneled through the VPN. I solved this in Windows by simply removing the SplitTunneling option in the Add-VPNConnection powershell and adding a route to my IP with the client's subnet gateway as next-hop. So my question is on the linux client, how do I use a remote_ts of 0.0.0.0/0 but exclude my IP from the policy so traffic to my IP does not get encrypted?

I've tried changing the linux client configuration to use xfrm interfaces but the challenge now is that I have to manually assign an IP to the xfrm interface. With a vip of 0.0.0.0 in swanctl, the client gets an IP but it is not assigned to the xfrm interface. Rather it gets assigned to the eth0 interface and traffic doesn't flow properly even though the VPN connects. I think if I assigned the IP manually, remove the vip option, and create routes I should be able to achieve the right flow but my implementation relies on the server assigning the IP address.

Appreciate any ideas.
Thanks

**Comment by tobiasbrunner** [comment, marked as answer]
You could change the interface on which virtual IPs are installed via `charon.install_virtual_ip_on`, but it doesn't really matter as long as the route via XFRM interface specifies the IP as preferred source address (you can install such a route via updown/vici once the virtual IP is known, or you can use the latest release and enable `charon.plugins.kernel-netlink.install_routes_xfrmi`).

Also, when not using XFRM interfaces, you could exclude remote IPs via [bypass policies](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies) (excluding protocols won't work with virtual IPs as that's forced as source via a route in table 220).

**Reply by ptchuba** [reply]
Thank you very much!
